vulcain.go

```diff
@@ -225,8 +225,16 @@ func (v *Vulcain) Apply(req *http.Request, rw http.ResponseWriter, responseBody
 	if fieldsHeader {
 		responseHeaders.Add("Vary", "Fields")
 	}
-	if addPreloadToVary {
+	if usePreloadLinks {
 		responseHeaders.Add("Vary", "Preload")
+
+		// forward preload links added by push() to a 103 Early Hints response
+		for _, link := range responseHeaders.Values("Link") {
+			rw.Header().Add("Link", link)
+		}
+		rw.WriteHeader(http.StatusEarlyHints)
+		// remove extra Link from final response headers
+		rw.Header().Del("Link")
```

Comment:
Why is this needed? I'm not sure to well understand what's going on here.

Reply:
https://github.com/dunglas/vulcain/pull/95#discussion_r1002766108

> Without this, the final response ends up with an extra - duplicated preload link. If you want to try it out (would be nice): checkout my PR, comment that line and run go test -v -timeout 30s -run PreloadHeader

Btw, should the final response actually have links if we already sent them in a 103?

Reply:
Yes, it even must have them, or downloaded data will be discarded.